compiler/rustc_resolve/src/lib.rs

```diff
@@ -961,6 +978,9 @@ pub struct Resolver<'a, 'tcx> {
     /// All non-determined imports.
     indeterminate_imports: Vec<Import<'a>>,
 
+    /// Imports for redundant check.
+    potentially_redundant_imports: FxHashMap<NodeId, Import<'a>>,
```

Comment:
On a second thought, it doesn't even make sense to collect these imports to the table - the table will contain almost all imports anyway.

It is simpler to organize a loop over all imports (see `check_hidden_glob_reexports` for an example) at the end of `fn check_unused`.